planning/behavior_path_planner/src/scene_module/avoidance_by_lc/module.cpp

```diff
@@ -0,0 +1,1380 @@
+using motion_utils::findNearestSegmentIndex;
+using tier4_autoware_utils::calcDistance2d;
+using tier4_autoware_utils::calcLateralDeviation;
+
+AvoidanceByLCModule::AvoidanceByLCModule(
+  const std::string & name, rclcpp::Node & node,
+  const std::shared_ptr<AvoidanceByLCParameters> parameters,
+  const std::unordered_map<std::string, std::shared_ptr<RTCInterface>> & rtc_interface_ptr_map)
+: SceneModuleInterface{name, node, rtc_interface_ptr_map}, parameters_{parameters}
+{
+  steering_factor_interface_ptr_ =
+    std::make_unique<SteeringFactorInterface>(&node, "avoidance_by_lane_change");
+}
+
+void AvoidanceByLCModule::processOnEntry()
+{
+#ifndef USE_OLD_ARCHITECTURE
+  waitApproval();
+#endif
+  current_lane_change_state_ = LaneChangeStates::Normal;
+  updateLaneChangeStatus();
+}
+
+void AvoidanceByLCModule::processOnExit()
+{
+  resetParameters();
+}
+
+bool AvoidanceByLCModule::isExecutionRequested() const
+{
+  if (current_state_ == ModuleStatus::RUNNING) {
+    return true;
+  }
+
+#ifdef USE_OLD_ARCHITECTURE
+  const auto current_lanes = util::getCurrentLanes(planner_data_);
+#else
+  const auto current_lanes =
+    util::getCurrentLanesFromPath(*getPreviousModuleOutput().reference_path, planner_data_);
+#endif
+  const auto lane_change_lanes = getLaneChangeLanes(current_lanes, lane_change_lane_length_);
+
+  LaneChangePath selected_path;
+  const auto [found_valid_path, found_safe_path] =
+    getSafePath(lane_change_lanes, check_distance_, selected_path);
+
+  if (!found_safe_path) {
+    return false;
+  }
+
+  const auto object_num = avoidance_data_.target_objects.size();
+  if (parameters_->execute_object_num > object_num) {
+    return false;
+  }
+
+  const auto to_front_object_distance = avoidance_data_.target_objects.front().longitudinal;
+  if (parameters_->execute_object_longitudinal_margin > to_front_object_distance) {
+    return false;
+  }
+
+  const auto to_lane_change_end_distance = calcSignedArcLength(
+    selected_path.path.points, getEgoPose().position, selected_path.shift_line.end.position);
+  const auto lane_change_finish_before_object =
+    to_front_object_distance > to_lane_change_end_distance;
+  if (
+    !lane_change_finish_before_object &&
+    parameters_->execute_only_when_lane_change_finish_before_object) {
+    return false;
+  }
+
+  return true;
+}
+
+bool AvoidanceByLCModule::isExecutionReady() const
+{
+  if (current_state_ == ModuleStatus::RUNNING) {
+    return true;
+  }
+
+#ifdef USE_OLD_ARCHITECTURE
+  const auto current_lanes = util::getCurrentLanes(planner_data_);
+#else
+  const auto current_lanes =
+    util::getCurrentLanesFromPath(*getPreviousModuleOutput().reference_path, planner_data_);
+#endif
+  const auto lane_change_lanes = getLaneChangeLanes(current_lanes, lane_change_lane_length_);
+
+  LaneChangePath selected_path;
+  const auto [found_valid_path, found_safe_path] =
+    getSafePath(lane_change_lanes, check_distance_, selected_path);
+
+  return found_valid_path;
+}
```

Comment:
in lane change module, `isExecutionReady()` depends on `found_safe_path`.
May I know why `AvoidanceByLC` checks for `found_valid_path` instead?

Reply:
Maybe that's my mistake. It is okay with `found_safe_path`.